src/functions.php

```diff
@@ -283,8 +283,8 @@ function awaitAll(array $promises, LoopInterface $loop, $timeout = null)
 function _cancelAllPromises(array $promises)
 {
     foreach ($promises as $promise) {
-        if ($promise instanceof CancellablePromiseInterface) {
-            $promise->cancel();
+        if ($promise instanceof PromiseInterface) {
+            Promise\resolve($promise)->cancel();
```

Comment:
Interesting! I see why this was done, but it looks like static analysis complains about this because `resolve()` has a `PromiseInterface` return type for Promise v2 and Promise v1.

Perhaps use logic similar to https://github.com/reactphp/promise-timer/pull/37/files#diff-fe65dcdace9cc44252b537bee79dd574edd1bccf6cee646cc860006a6ec50e8bR15? /cc @WyriHaximus

What do you think about this?

Reply:
> Perhaps use logic similar to https://github.com/reactphp/promise-timer/pull/37/files#diff-fe65dcdace9cc44252b537bee79dd574edd1bccf6cee646cc860006a6ec50e8bR15? /cc @WyriHaximus

That is the preferred way, making it more explicit, and more performant.

Reply:
Yeah I thought so, had this initially but changed it due to [this comment](https://github.com/reactphp/promise/pull/75#issuecomment-270373856). I'll change it back as it's more lightweight.